packages/firestore/lib/src/collection_reference.dart

```diff
@@ -0,0 +1,51 @@
+// Copyright 2017, the Flutter project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+part of firebase_firestore;
+
+/// A CollectionReference object can be used for adding documents, getting
+/// document references, and querying for documents (using the methods
+/// inherited from Query).
+class CollectionReference extends Query {
+  CollectionReference._(Firestore firestore, List<String> pathComponents)
+      : super._(firestore: firestore, pathComponents: pathComponents);
+
+  /// For subcollections, parent returns the containing DocumentReference.
+  ///
+  /// For root collections, null is returned.
+  CollectionReference parent() {
+    if (_pathComponents.isEmpty) {
+      return null;
+    }
+    return new CollectionReference._(
+        _firestore, (new List<String>.from(_pathComponents)..removeLast()));
+  }
+
+  String get key => _pathComponents.last;
```

Comment:
Document this public member?

Reply:
Good catch, it should actually be path. I'll fix.